scipy/_lib/_array_api.py

```diff
@@ -215,12 +216,9 @@ def xp_copy(x: Array, *, xp: ModuleType | None = None) -> Array:
     This copy function does not offer all the semantics of `np.copy`, i.e. the
     `subok` and `order` keywords are not used.
     """
-    # Note: for older NumPy versions, `np.asarray` did not support the `copy` kwarg,
-    # so this uses our other helper `_asarray`.
     if xp is None:
         xp = array_namespace(x)
-
-    return _asarray(x, copy=True, xp=xp)
+    return xp.asarray(x, copy=True)
```

Comment:
can we rely on this working for `np<2`? Or are there no codepaths where `xp` is the raw NumPy namespace?

Reply:
I think that `xp` here is `array_api_compat.numpy`, which handles it.
Unit tests will confirm/dismiss my understanding when they finish.

Reply:
...it is not. reverting.